src/util/fips.js

```diff
@@ -0,0 +1,85 @@
+/* Copyright (C) 2016 NooBaa */
+'use strict';
+
+const fs = require('fs');
+const crypto = require('crypto');
+const stream = require('stream');
+const nb_native = require('./nb_native');
+
+const original_crypto = Object.freeze({ ...crypto });
+let fips_mode = false;
+
+/**
+ * @returns {Boolean}
+ */
+function get_fips_mode() {
+    return fips_mode;
+}
+
+class HashWrap extends stream.Transform {
+    constructor(hash) {
+        super();
+        this.hash = hash;
+    }
+    update(data, encoding) {
+        return this.hash.update(encoding ? Buffer.from(data, encoding) : data);
+    }
+    digest(encoding) {
+        const buf = this.hash.digest();
+        return encoding ? buf.toString(encoding) : buf;
+    }
+    _transform(data, encoding, next) {
+        this.update(data, encoding);
+        next();
+    }
+    copy() {
+        // unimplemented
+        return null;
+    }
+}
+
+/**
+ * @param {Boolean} mode
+ */
+function set_fips_mode(mode = detect_fips_mode()) {
+    nb_native().set_fips_mode(mode);
+    if (mode) {
+        // monkey-patch the crypto.createHash() function to provide a non-crypto md5 flow
+        crypto.createHash = function(algorithm, options) {
+            switch (algorithm) {
+                case 'md5': {
+                    return new HashWrap(new(nb_native().MD5_MB)());
+                }
+                default:
+                    return original_crypto.createHash(algorithm, options);
+            }
+        };
+    } else if (crypto.createHash !== original_crypto.createHash) {
+        crypto.createHash = original_crypto.createHash;
+    }
+}
+
+
+/**
+ * @returns {Boolean}
+ */
+function detect_fips_mode() {
+    if (process.env.FIPS) return true;
+    const fips_proc_file = '/proc/crypto/fips';
+    try {
+        const value = fs.readFileSync(fips_proc_file, 'utf8').trim();
+        console.log(`detect_fips_mode: found ${fips_proc_file} with value ${value}`);
+        return value === '1';
+    } catch (err) {
+        if (err.code !== 'ENOENT') {
+            console.warn(`detect_fips_mode: failed to read ${fips_proc_file}:`, err);
```

Comment:
This will happen in most cases as centos:8 which the upstream based on have no /proc/crypto/fips 
We can check if the file even exist and console log it
What do you think?

Reply:
resolved by excluding the print on ENOTDIR.